cleverpush/src/main/java/com/cleverpush/stories/StoryDetailActivity.java

```diff
@@ -178,94 +219,122 @@ protected void onPause() {
-          }
+        public void onPageFinished(WebView view, String url) {
+          super.onPageFinished(view, url);
         }
       });
-
+      String htmlContent = loadHtml();
+      webView.loadDataWithBaseURL(null, htmlContent, "text/html; charset=utf-8", "UTF-8", null);
     } catch (Exception e) {
       Logger.e(TAG, "Error loading story details in StoryDetailActivity", e);
     }
   }
 
-  @Override
-  public void onNext(int position) {
-    try {
-      if (position != stories.size() - 1) {
-        selectedPosition = position + 1;
-
-        if (storyDetailListAdapter != null) {
-          int subStoryIndex = getSubStoryPosition();
-          storyDetailListAdapter.subStoryPosition = subStoryIndex;
-          storyDetailListAdapter.selectedPosition = selectedPosition;
-          storyDetailListAdapter.apiCallMap.put(selectedPosition, false);
-          storyDetailListAdapter.refresh(selectedPosition, subStoryIndex);
-        }
+  public String loadHtml() {
+    StringBuilder anchorTags = new StringBuilder();
 
-        recyclerView.smoothScrollToPosition(selectedPosition);
-
-        String storyId = stories.get(position + 1).getId();
-        setStoryOpened(storyId);
-        trackStoryOpened();
+    for (int i = 0; i < stories.size(); i++) {
+      String storyId = stories.get(i).getId();
+      String customURL = "";
+      int subStoryIndex = getSubStoryPosition(i);
+      if (stories.get(i).getContent().getPages() != null && stories.get(i).getContent().getPages().size() > 1) {
+        customURL = "https://api.cleverpush.com/channel/" + stories.get(i).getChannel() + "/story/" + storyId + "/html?hideStoryShareButton=" + isHideStoryShareButton + "&widgetId=" + widgetId + "&#page=page-" + subStoryIndex;
       } else {
-        runOnUiThread(this::finish);
+        customURL = "https://api.cleverpush.com/channel/" + stories.get(i).getChannel() + "/story/" + storyId + "/html?hideStoryShareButton=" + isHideStoryShareButton + "&widgetId=" + widgetId;
       }
-    } catch (Exception e) {
-      Logger.e(TAG, "Error handling onNext in StoryDetailActivity", e);
+      anchorTags.append("    <a href=\"").append(customURL).append("\">\"Story Title ").append(i + 1).append(" \"</a>\n");
     }
+
+    return "<!DOCTYPE html>\n" +
+        "<html>\n" +
+        "<head>\n" +
+        "  <script src=\"https://cdn.ampproject.org/amp-story-player-v0.js\"></script>\n" +
+        "  <link rel=\"stylesheet\" href=\"https://cdn.ampproject.org/amp-story-player-v0.css\">\n" +
+        "  <style>\n" +
+        "    body { margin: 0; padding: 0; }\n" +
+        "    amp-story-player { display: block; margin: 0; padding: 0; width: 100%%; height: " + convertPixelsToDp(measuredHeight, getApplicationContext()) + "px; }\n" +
+        "  </style>\n" +
+        "</head>\n" +
+        "<body>\n" +
+        "  <amp-story-player width=\"100%%\" height=\"" + convertPixelsToDp(measuredHeight, getApplicationContext()) + "\">\n" +
```

Comment:
Remove width and height here, so it's only `<amp-story-player>` ( we handle the styles above )